pkg/route/ingress/ingress.go

```diff
@@ -405,6 +409,24 @@ func (c *Controller) sync(key queueKey) error {
 			}
 
 			if routeMatchesIngress(existing, ingress, &rule, &path, c.secretLister, c.serviceLister) {
+				var loadbalancerIngresses []corev1.LoadBalancerIngress
+				for _, rule := range ingress.Spec.Rules {
```

Comment:
Shouldn't this be using the status of the route?  I would expect the logic to use `old` or `existing` and collect `old[i].Status.Ingress[j].RouterCanonicalHostname` for the ingress status.  If we take the approach that I am suggesting, the logic should also uniquify the list, or use `sets.String` when collecting names, because one ingress may have many routes that are all exposed by the same router.

Would it make more sense to do this before the loop, on line 387?  We already have the list of routes that correspond to the ingress (as indicated by their owner references) there; putting the new logic here actually seems to complicate things.

The semantics gets iffy if, for example, an ingress is translated into multiple routes, one of those routes is admitted by some but not all routers, and another route is admitted by some router that did not admit the first route.  However, I think collecting all canonical names of all routers that expose at least one of the ingress's routes is the most reasonable translation within the constraints of the API.